lib/improver/tests/lapse_rate/Test_LapseRate.py

```diff
@@ -0,0 +1,367 @@
+                                 [0.0294, 0.0294, 0.0, -0.0098, -0.0098],
+                                 [0.0, 0.0, 0.0, 0.0, 0.0],
+                                 [0.0, 0.0, 0.0, 0.0, 0.0]])
+
+        # West data points should be -3*DALR and East should be DALR, South
+        # should be zero.
+        self.temperature.data[:, :, 0:2] = 1
+        self.temperature.data[:, :, 3:5] = -1
+        self.orography.data[:, :] = 10
+        self.land_sea_mask.data[3:5, :] = 0
+
+        result = LapseRate().process(self.temperature, self.orography,
+                                     self.land_sea_mask)
+
+        self.assertArrayAlmostEqual(result.data, expected_out)
+
+    def test_decr_temp_incr_orog(self):
+        """ Test code where temperature is decreasing with height. This is the
+            expected scenario for lapse rate.
+        """
+        reset_cube_data(self.temperature, self.orography, self.land_sea_mask)
+
+        expected_out = np.array([[0.0, -0.0098, -0.0098, -0.00642857, -0.005],
+                                 [0.0, -0.0098, -0.0098, -0.00642857, -0.005],
+                                 [0.0, -0.0098, -0.0098, -0.00642857, -0.005],
+                                 [0.0, -0.0098, -0.0098, -0.00642857, -0.005],
+                                 [0.0, -0.0098, -0.0098, -0.00642857, -0.005]])
+
+        self.temperature.data[:, :, 0:2] = 0.4
+        self.temperature.data[:, :, 2] = 0.3
+        self.temperature.data[:, :, 3] = 0.2
+        self.temperature.data[:, :, 4] = 0.1
+
+        self.orography.data[:, 2] = 10
+        self.orography.data[:, 3] = 20
+        self.orography.data[:, 4] = 40
+
+        result = LapseRate().process(self.temperature, self.orography,
+                                     self.land_sea_mask)
+
+        self.assertArrayAlmostEqual(result.data, expected_out)
+
+    def test_decr_temp_decr_orog(self):
+        """ Test code where the temperature increases with height.
+        """
+        reset_cube_data(self.temperature, self.orography, self.land_sea_mask)
+
+        expected_out = np.array([[0.0, 0.01, 0.01, 0.00642857, 0.005],
+                                 [0.0, 0.01, 0.01, 0.00642857, 0.005],
+                                 [0.0, 0.01, 0.01, 0.00642857, 0.005],
+                                 [0.0, 0.01, 0.01, 0.00642857, 0.005],
+                                 [0.0, 0.01, 0.01, 0.00642857, 0.005]])
+
+        self.temperature.data[:, :, 0:2] = 0.1
+        self.temperature.data[:, :, 2] = 0.2
+        self.temperature.data[:, :, 3] = 0.3
+        self.temperature.data[:, :, 4] = 0.4
+
+        self.orography.data[:, 2] = 10
+        self.orography.data[:, 3] = 20
+        self.orography.data[:, 4] = 40
+
+        result = LapseRate().process(self.temperature, self.orography,
+                                     self.land_sea_mask)
+
+        self.assertArrayAlmostEqual(result.data, expected_out)
+
```

Comment:
Could we have an additional test for "process" please, to cover the situation where some neighbourhoods will have points masked out due to exceeding the maximum height difference?